packages/calcite-components/src/components/input/assets/_mixins.scss

```diff
@@ -0,0 +1,819 @@
+@mixin setOnInputEl($type) {
+  @if ($type == "*") {
+    textarea,
+    input {
+      @content;
+    }
+  } @else if ($type == "textarea") {
+    textarea {
+      @content;
+    }
+  } @else {
+    input {
+      @content;
+    }
+  }
+}
+
+@mixin setInputTokens($component: "input", $subcomponent: "") {
+  // Set Defaults
+  @if ($subcomponent == "") {
+    :host {
+      --calcite-#{$component}-corner-radius: var(--calcite-corner-radius);
```

Comment:
This still doesn't solve the issue that if filter does the following:

```scss
calcite-input {
  --calcite-input-corner-radius: var(--calcite-filter-input-corner-radius);
}
```

The input's corner radius will be set to undefined unless a user explicitly sets it.

Input would need to set a fallback for these values or filter, and any other component that uses input internally would either have to copy the default from input and add it as a fallback. IMO it makes more sense for the fallback to be in the component that is defining the token.